### PR TITLE
[TS] LPS-111067 Make Display Style persist in multiple menus

### PR DESCRIPTION
**Pull request by izaera**
Refactor of https://github.com/liferay-frontend/liferay-portal/pull/753

----

This is being sent here per @jbalsas [here](https://github.com/liferay/liferay-portal-ee/pull/22571#issuecomment-768448172).

https://issues.liferay.com/browse/LPS-111067

The issue is essentially that our current pattern used when storing the displayStyle of a portlet causes it to be lost when switching between portlets since the request parameter doesn't persist. So, for example, if I prefer to view my site memberships as all lists (and switch my display style for each to "list"), then switch to upload a document, when I return to my site memberships they will have all reverted back to their default view style (which is "icon" in some cases).

This fix adds the previous pattern, which stored the displayStyle in portalPreferences, as a backup to the current pattern so that we can persist between portlets/sessions. This logic actually appears in a handful of portlets already (see: https://github.com/liferay/liferay-portal/blame/master/modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/display/context/BlogEntriesDisplayContext.java#L123-L131), so this can be seen as simply applying it more broadly.

Please let me know if you have any questions. Thanks!

### CODE REVIEW

**Review by jbalsas — APPROVED** [review]
LGTM @izaera! Is baseline happy with this?

@wanderlast, could you validate if this is still working as expected before we forward it?

Thanks!

**Comment by izaera** [comment]
> Is baseline happy with this?

Not sure. Shouldn't `ci:test:sf` have detected it? :thinking: 

I got contradictory results with local SF...

**Comment by jbalsas** [comment]
> Not sure. Shouldn't ci:test:sf have detected it? 🤔
>
> I got contradictory results with local SF...

I don't think so, since it's a kernel change...

**Comment by wincent** [comment]
@ealonso @cgoncas @csierra @adolfopa: This is the PR I mentioned in our sync meeting that impacts modules belonging to your teams.

**Comment by izaera** [comment]
> I don't think so, since it's a kernel change...

Setting PR to draft until I can fix it (I can't switch branches now)

File: modules/apps/mobile-device-rules/mobile-device-rules-web/src/main/java/com/liferay/mobile/device/rules/web/internal/display/context/MDRActionDisplayContext.java [hunk 1]
```diff
@@ -88,8 +92,18 @@ public String getDisplayStyle() {
...
+
+		PortalPreferences portalPreferences =
+			PortletPreferencesFactoryUtil.getPortalPreferences(_renderRequest);
+
+		if (Validator.isNull(_displayStyle)) {
+			_displayStyle = portalPreferences.getValue(
+				MDRPortletKeys.MOBILE_DEVICE_RULES, "display-style", "list");
+		}
+
+		portalPreferences.setValue(
```
Comment (shuyangzhou) [inline modules/apps/mobile-device-rules/mobile-device-rules-web/src/main/java/com/liferay/mobile/device/rules/web/internal/display/context/MDRActionDisplayContext.java]: Shouldn't this be inside a "else"? If you were just inside the "if (Validator.isNull(_displayStyle)) {" you just got the "_displayStyle" from portalPreferences, why would you want to put the same value back in?

Comment (izaera) [reply modules/apps/mobile-device-rules/mobile-device-rules-web/src/main/java/com/liferay/mobile/device/rules/web/internal/display/context/MDRActionDisplayContext.java]: :thinking: That's changed in the final commits. Perhaps GitHub is playing with your display and missed some commits?

Comment (shuyangzhou) [reply modules/apps/mobile-device-rules/mobile-device-rules-web/src/main/java/com/liferay/mobile/device/rules/web/internal/display/context/MDRActionDisplayContext.java]: Oh, ok, I was just given this single commit to review, I did not check the whole pull.

Comment (brunofernandezg) [reply modules/apps/mobile-device-rules/mobile-device-rules-web/src/main/java/com/liferay/mobile/device/rules/web/internal/display/context/MDRActionDisplayContext.java]: Thanks @shuyangzhou, it was a miscommunication :worried:

Comment (izaera) [reply modules/apps/mobile-device-rules/mobile-device-rules-web/src/main/java/com/liferay/mobile/device/rules/web/internal/display/context/MDRActionDisplayContext.java]: :+1: No worries. GitHub has these things...

**Comment by jbalsas** [comment]
Looks like baseline is okay. I'm going to retest to see if those are flukes or need some test fixes. It might happen that we're persisting now some displayStyle from one run to another and then test selectors fail... (hope it's just a fluke, though).

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by izaera** [comment]
> Looks like baseline is okay

Why? Because the tests didn't fail? I understood the opposite from your comment yesterday :thinking:

**Comment by jbalsas** [comment]
> Why? Because the tests didn't fail? I understood the opposite from your comment yesterday 🤔

Yeah, we should see a baseline error in the tests if there was a baseline issue.

**Comment by izaera** [comment]
Checking if failures are legit...

**Comment by izaera** [comment]
ci:test:sf

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
This is working now. Can we forward it to Brian? 

@shuyangzhou , @wanderlast ?

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**